### ISSUE TITLE
Timeseries classification from scratch's inference result shape is different from what I expected. 

### ISSUE THREAD

**Issue by pstar0830** [keras-team/keras-io#1929]
### Issue Type

Others

### Source

source

### Keras Version

3.0.5

### Custom Code

Yes

### OS Platform and Distribution

Ubuntu 22.04.1 

### Python version

3.10

### Current Behavior?

I am following the tutorial below.
- https://keras.io/examples/timeseries/timeseries_classification_from_scratch/
- https://github.com/keras-team/keras-io/blob/master/examples/timeseries/timeseries_classification_from_scratch.py

The tutorial has training and evaluation code, but not inference code, so I wrote an inference code at the end of the code to test individual inference. 

```
res = model.predict(x_test[0])  # x_test[0].shape : (500, 1)
res.shape # is (500, 2)
```

I can only use single input for the predict() (not batch input), I guess due to `make_model(input_shape=x_train.shape[1:])`
what I don't understand is that the res.shape is (500, 2), not (2).

model.input_shape is (None, 500, 1) and model.output_shape is (None, 2)

I used (500, 1) shaped input, but why I got (500, 2) shaped output? 
Could someone tell me what I did wrong here?

Thank you in advance.

### Standalone code to reproduce the issue or tutorial link

```shell
https://keras.io/examples/timeseries/timeseries_classification_from_scratch/

res = model.predict(x_test[0])  # x_test[0].shape : (500, 1)
res.shape # is (500, 2)
```

**Comment by pstar0830** [comment]
when I use `mode(x_test)`, it works fine. 
don't know why predict() doesn't work..

**Comment by dhantule** [comment]
Hi @pstar0830, 

Since model's input_shape is `(None, 500, 1)` expects input of shape : `(batch_size, timesteps, features)` when you pass 
`res = model.predict(x_test[0])` it has no batch dimension to correctly predict for a single input try using

`res = model.predict(np.expand_dims(x_test[0], axis=0))  # Shape: (1, 500, 1)`
